src/Data/Number.js

```diff
@@ -0,0 +1,10 @@
+exports.fromStringImpl = function (just) {
+  return function (nothing) {
+    return function (string) {
+      var result = parseFloat(string);
+
+      return isNaN(result) ? nothing
```

Comment:
Agreed on `Infinity`, will amend now. Interestingly, PureScript's core libraries tend to take the stance that NaN/Infinity is "undefined" behaviour. I wonder what @paf31 would think of `isNaN` and `isFinite` existing in PS.